api/v1beta1/azurecluster_webhook.go

```diff
@@ -31,29 +33,50 @@ import (
 
 // SetupWebhookWithManager sets up and registers the webhook with the manager.
 func (c *AzureCluster) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	w := new(AzureClusterWebhook)
 	return ctrl.NewWebhookManagedBy(mgr).
 		For(c).
+		WithValidator(w).
+		WithDefaulter(w).
 		Complete()
 }
 
 // +kubebuilder:webhook:verbs=create;update,path=/validate-infrastructure-cluster-x-k8s-io-v1beta1-azurecluster,mutating=false,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azureclusters,versions=v1beta1,name=validation.azurecluster.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
 // +kubebuilder:webhook:verbs=create;update,path=/mutate-infrastructure-cluster-x-k8s-io-v1beta1-azurecluster,mutating=true,failurePolicy=fail,matchPolicy=Equivalent,groups=infrastructure.cluster.x-k8s.io,resources=azureclusters,versions=v1beta1,name=default.azurecluster.infrastructure.cluster.x-k8s.io,sideEffects=None,admissionReviewVersions=v1;v1beta1
 
-var _ webhook.Validator = &AzureCluster{}
-var _ webhook.Defaulter = &AzureCluster{}
+type AzureClusterWebhook struct{}
+
+var _ webhook.CustomValidator = &AzureClusterWebhook{}
+var _ webhook.CustomDefaulter = &AzureClusterWebhook{}
+
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type.
+func (_ *AzureClusterWebhook) Default(_ context.Context, obj runtime.Object) error {
```

Comment:
```suggestion
func (*AzureClusterWebhook) Default(_ context.Context, obj runtime.Object) error {
```

The linter complains about `_` as a receiver name and suggests omitting it.